File: app/src/main/java/com/vondear/tools/activity/ActivityRxDataTool.java
package com.vondear.tools.activity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.vondear.rxtools.RxDataTool;
import com.vondear.rxtools.activity.ActivityBase;
import com.vondear.rxtools.view.RxTitle;
import com.vondear.tools.R;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * @author vondear
 */
public class ActivityRxDataTool extends ActivityBase {

    @BindView(R.id.editText)
    EditText mEditText;
    @BindView(R.id.btn_null)
    Button mBtnNull;
    @BindView(R.id.tv_null)
    TextView mTvNull;
    @BindView(R.id.btn_null_obj)
    Button mBtnNullObj;
    @BindView(R.id.tv_null_obj)
    TextView mTvNullObj;
    @BindView(R.id.btn_is_integer)
    Button mBtnIsInteger;
    @BindView(R.id.tv_is_integer)
    TextView mTvIsInteger;
    @BindView(R.id.btn_is_double)
    Button mBtnIsDouble;
    @BindView(R.id.tv_is_double)
    TextView mTvIsDouble;
    @BindView(R.id.btn_is_number)
    Button mBtnIsNumber;
    @BindView(R.id.tv_is_number)
    TextView mTvIsNumber;
    @BindView(R.id.ed_month)
    EditText mEdMonth;
    @BindView(R.id.ed_day)
    EditText mEdDay;
    @BindView(R.id.btn_astro)
    Button mBtnAstro;
    @BindView(R.id.tv_astro)
    TextView mTvAstro;
    @BindView(R.id.ed_mobile)
    EditText mEdMobile;
    @BindView(R.id.btn_hind_mobile)
    Button mBtnHindMobile;
    @BindView(R.id.tv_hind_mobile)
    TextView mTvHindMobile;
    @BindView(R.id.ed_bank_card)
    EditText mEdBankCard;
    @BindView(R.id.btn_format_bank_card)
    Button mBtnFormatBankCard;
    @BindView(R.id.tv_format_bank_card)
    TextView mTvFormatBankCard;
    @BindView(R.id.btn_format_bank_card_4)
    Button mBtnFormatBankCard4;
    @BindView(R.id.tv_format_bank_card_4)
    TextView mTvFormatBankCard4;
    @BindView(R.id.btn_getAmountValue)
    Button mBtnGetAmountValue;
    @BindView(R.id.tv_getAmountValue)
    TextView mTvGetAmountValue;
    @BindView(R.id.btn_getRoundUp)
    Button mBtnGetRoundUp;
    @BindView(R.id.tv_getRoundUp)
    TextView mTvGetRoundUp;
    @BindView(R.id.ed_text)
    EditText mEdText;
    @BindView(R.id.btn_string_to_int)
    Button mBtnStringToInt;
    @BindView(R.id.tv_string_to_int)
    TextView mTvStringToInt;
    @BindView(R.id.btn_string_to_long)
    Button mBtnStringToLong;
    @BindView(R.id.tv_string_to_long)
    TextView mTvStringToLong;
    @BindView(R.id.btn_string_to_double)
    Button mBtnStringToDouble;
    @BindView(R.id.tv_string_to_double)
    TextView mTvStringToDouble;
    @BindView(R.id.btn_string_to_float)
    Button mBtnStringToFloat;
    @BindView(R.id.tv_string_to_float)
    TextView mTvStringToFloat;
    @BindView(R.id.btn_string_to_two_number)
    Button mBtnStringToTwoNumber;
    @BindView(R.id.tv_string_to_two_number)
    TextView mTvStringToTwoNumber;
    @BindView(R.id.btn_upperFirstLetter)
    Button mBtnUpperFirstLetter;
    @BindView(R.id.tv_upperFirstLetter)
    TextView mTvUpperFirstLetter;
    @BindView(R.id.btn_lowerFirstLetter)
    Button mBtnLowerFirstLetter;
    @BindView(R.id.tv_lowerFirstLetter)
    TextView mTvLowerFirstLetter;
    @BindView(R.id.btn_reverse)
    Button mBtnReverse;
    @BindView(R.id.tv_reverse)
    TextView mTvReverse;
    @BindView(R.id.btn_toDBC)
    Button mBtnToDBC;
    @BindView(R.id.tv_toDBC)
    TextView mTvToDBC;
    @BindView(R.id.btn_toSBC)
    Button mBtnToSBC;
    @BindView(R.id.tv_toSBC)
    TextView mTvToSBC;
    @BindView(R.id.btn_oneCn2ASCII)
    Button mBtnOneCn2ASCII;
    @BindView(R.id.tv_oneCn2ASCII)
    TextView mTvOneCn2ASCII;
    @BindView(R.id.btn_oneCn2PY)
    Button mBtnOneCn2PY;
    @BindView(R.id.tv_oneCn2PY)
    TextView mTvOneCn2PY;
    @BindView(R.id.btn_getPYFirstLetter)
    Button mBtnGetPYFirstLetter;
    @BindView(R.id.tv_getPYFirstLetter)
    TextView mTvGetPYFirstLetter;
    @BindView(R.id.btn_cn2PY)
    Button mBtnCn2PY;
    @BindView(R.id.tv_cn2PY)
    TextView mTvCn2PY;
    @BindView(R.id.ed_money)
    EditText mEdMoney;
    @BindView(R.id.ed_string)
    EditText mEdString;
    @BindView(R.id.rx_title)
    RxTitle mRxTitle;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_rx_data_utils);
        ButterKnife.bind(this);
        mRxTitle.setLeftFinish(mContext);
    }

    @OnClick({R.id.btn_null, R.id.btn_null_obj, R.id.btn_is_integer, R.id.btn_is_double, R.id.btn_is_number, R.id.btn_astro, R.id.btn_hind_mobile, R.id.btn_format_bank_card, R.id.btn_format_bank_card_4, R.id.btn_getAmountValue, R.id.btn_getRoundUp, R.id.btn_string_to_int, R.id.btn_string_to_long, R.id.btn_string_to_double, R.id.btn_string_to_float, R.id.btn_string_to_two_number, R.id.btn_upperFirstLetter, R.id.btn_lowerFirstLetter, R.id.btn_reverse, R.id.btn_toDBC, R.id.btn_toSBC, R.id.btn_oneCn2ASCII, R.id.btn_oneCn2PY, R.id.btn_getPYFirstLetter, R.id.btn_cn2PY})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.btn_null:
                mTvNull.setText(RxDataTool.isNullString(mEditText.getText().toString()) + "");
                break;
            case R.id.btn_null_obj:
                mTvNullObj.setText(RxDataTool.isNullString(mEditText.getText().toString()) + "");
                break;
            case R.id.btn_is_integer:
                mTvIsInteger.setText(RxDataTool.isInteger(mEditText.getText().toString()) + "");
                break;
            case R.id.btn_is_double:
                mTvIsDouble.setText(RxDataTool.isDouble(mEditText.getText().toString()) + "");
                break;
            case R.id.btn_is_number:
                mTvIsNumber.setText(RxDataTool.isNumber(mEditText.getText().toString()) + "");
                break;
            case R.id.btn_astro:
                mTvAstro.setText(RxDataTool.getAstro(RxDataTool.stringToInt(mEdMonth.getText().toString()), RxDataTool.stringToInt(mEdDay.getText().toString())));
                break;
            case R.id.btn_hind_mobile:
                mTvHindMobile.setText(RxDataTool.hideMobilePhone4(mEdMobile.getText().toString()));
                break;
            case R.id.btn_format_bank_card:
                mTvFormatBankCard.setText(RxDataTool.formatCard(mEdBankCard.getText().toString()));
                break;
            case R.id.btn_format_bank_card_4:
                mTvFormatBankCard4.setText(RxDataTool.formatCardEnd4(mEdBankCard.getText().toString()));
                break;
            case R.id.btn_getAmountValue:
                mTvGetAmountValue.setText(RxDataTool.getAmountValue(mEdMoney.getText().toString()));
                break;
            case R.id.btn_getRoundUp:
                mTvGetRoundUp.setText(RxDataTool.getRoundUp(mEdMoney.getText().toString(), 2));
                break;
            case R.id.btn_string_to_int:
                mTvStringToInt.setText(RxDataTool.stringToInt(mEdText.getText().toString()) + "");
                break;
            case R.id.btn_string_to_long:
                mTvStringToLong.setText(RxDataTool.stringToLong(mEdText.getText().toString()) + "");
                break;
            case R.id.btn_string_to_double:
                mTvStringToDouble.setText(RxDataTool.stringToDouble(mEdText.getText().toString()) + "");
                break;
            case R.id.btn_string_to_float:
                mTvStringToFloat.setText(RxDataTool.stringToFloat(mEdText.getText().toString()) + "");
                break;
            case R.id.btn_string_to_two_number:
                mTvStringToTwoNumber.setText(RxDataTool.format2Decimals(mEdText.getText().toString()) + "");
                break;
            case R.id.btn_upperFirstLetter:
                mTvUpperFirstLetter.setText(RxDataTool.upperFirstLetter(mEdString.getText().toString()));
                break;
            case R.id.btn_lowerFirstLetter:
                mTvLowerFirstLetter.setText(RxDataTool.lowerFirstLetter(mEdString.getText().toString()));
                break;
            case R.id.btn_reverse:
                mTvReverse.setText(RxDataTool.reverse(mEdString.getText().toString()));
                break;
            case R.id.btn_toDBC:
                mTvToDBC.setText(RxDataTool.toDBC(mEdString.getText().toString()));
                break;
            case R.id.btn_toSBC:
                mTvToSBC.setText(RxDataTool.toSBC(mEdString.getText().toString()));
                break;
            case R.id.btn_oneCn2ASCII:
                mTvOneCn2ASCII.setText(RxDataTool.oneCn2ASCII(mEdString.getText().toString()) + "");
                break;
            case R.id.btn_oneCn2PY:
                mTvOneCn2PY.setText(RxDataTool.oneCn2PY(mEdString.getText().toString()));
                break;
            case R.id.btn_getPYFirstLetter:
                mTvGetPYFirstLetter.setText(RxDataTool.getPYFirstLetter(mEdString.getText().toString()));
                break;
            case R.id.btn_cn2PY:
                mTvCn2PY.setText(RxDataTool.cn2PY(mEdString.getText().toString()));
                break;
        }
    }
}

File: RxDemo/src/main/java/com/tamsiree/rxdemo/activity/ActivityRxDataTool.kt
package com.tamsiree.rxdemo.activity

import android.annotation.SuppressLint
import android.os.Bundle
import com.tamsiree.rxdemo.R
import com.tamsiree.rxkit.RxDataTool.Companion.cn2PY
import com.tamsiree.rxkit.RxDataTool.Companion.format2Decimals
import com.tamsiree.rxkit.RxDataTool.Companion.formatCard
import com.tamsiree.rxkit.RxDataTool.Companion.formatCardEnd4
import com.tamsiree.rxkit.RxDataTool.Companion.getAmountValue
import com.tamsiree.rxkit.RxDataTool.Companion.getAstro
import com.tamsiree.rxkit.RxDataTool.Companion.getPYAllFirstLetter
import com.tamsiree.rxkit.RxDataTool.Companion.getPYFirstLetter
import com.tamsiree.rxkit.RxDataTool.Companion.getRoundUp
import com.tamsiree.rxkit.RxDataTool.Companion.hideMobilePhone4
import com.tamsiree.rxkit.RxDataTool.Companion.isDouble
import com.tamsiree.rxkit.RxDataTool.Companion.isInteger
import com.tamsiree.rxkit.RxDataTool.Companion.isNullString
import com.tamsiree.rxkit.RxDataTool.Companion.isNumber
import com.tamsiree.rxkit.RxDataTool.Companion.lowerFirstLetter
import com.tamsiree.rxkit.RxDataTool.Companion.oneCn2ASCII
import com.tamsiree.rxkit.RxDataTool.Companion.oneCn2PY
import com.tamsiree.rxkit.RxDataTool.Companion.reverse
import com.tamsiree.rxkit.RxDataTool.Companion.stringToDouble
import com.tamsiree.rxkit.RxDataTool.Companion.stringToFloat
import com.tamsiree.rxkit.RxDataTool.Companion.stringToInt
import com.tamsiree.rxkit.RxDataTool.Companion.stringToLong
import com.tamsiree.rxkit.RxDataTool.Companion.toDBC
import com.tamsiree.rxkit.RxDataTool.Companion.toSBC
import com.tamsiree.rxkit.RxDataTool.Companion.upperFirstLetter
import com.tamsiree.rxkit.RxDeviceTool.setPortrait
import com.tamsiree.rxui.activity.ActivityBase
import kotlinx.android.synthetic.main.activity_rx_data_utils.*

/**
 * @author tamsiree
 */
class ActivityRxDataTool : ActivityBase() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_rx_data_utils)
        setPortrait(this)
    }

    @SuppressLint("SetTextI18n")
    override fun initView() {
        rx_title.setLeftFinish(mContext)
        btn_null.setOnClickListener { tv_null.text = isNullString(editText.text.toString()).toString() }
        btn_null_obj.setOnClickListener { tv_null_obj.text = isNullString(editText.text.toString()).toString() }
        btn_is_integer.setOnClickListener { tv_is_integer.text = isInteger(editText.text.toString()).toString() }
        btn_is_double.setOnClickListener { tv_is_double.text = isDouble(editText.text.toString()).toString() }
        btn_is_number.setOnClickListener { tv_is_number.text = isNumber(editText.text.toString()).toString() }
        btn_astro.setOnClickListener { tv_astro.text = getAstro(stringToInt(ed_month.text.toString()), stringToInt(ed_day.text.toString())) }
        btn_hind_mobile.setOnClickListener { tv_hind_mobile.text = hideMobilePhone4(ed_mobile.text.toString()) }
        btn_format_bank_card.setOnClickListener { tv_format_bank_card.text = formatCard(ed_bank_card.text.toString()) }
        btn_format_bank_card_4.setOnClickListener { tv_format_bank_card_4.text = formatCardEnd4(ed_bank_card.text.toString()) }
        btn_getAmountValue.setOnClickListener { tv_getAmountValue.text = getAmountValue(ed_money.text.toString()) }
        btn_getRoundUp.setOnClickListener { tv_getRoundUp.text = getRoundUp(ed_money.text.toString(), 2) }
        btn_string_to_int.setOnClickListener { tv_string_to_int.text = stringToInt(ed_text.text.toString()).toString() }
        btn_string_to_long.setOnClickListener { tv_string_to_long.text = stringToLong(ed_text.text.toString()).toString() }
        btn_string_to_double.setOnClickListener { tv_string_to_double.text = stringToDouble(ed_text.text.toString()).toString() }
        btn_string_to_float.setOnClickListener { tv_string_to_float.text = stringToFloat(ed_text.text.toString()).toString() }
        btn_string_to_two_number.setOnClickListener { tv_string_to_two_number.text = format2Decimals(ed_text.text.toString()) }
        btn_upperFirstLetter.setOnClickListener { tv_upperFirstLetter.text = upperFirstLetter(ed_string.text.toString()) }
        btn_lowerFirstLetter.setOnClickListener { tv_lowerFirstLetter.text = lowerFirstLetter(ed_string.text.toString()) }
        btn_reverse.setOnClickListener { tv_reverse.text = reverse(ed_string.text.toString()) }
        btn_toDBC.setOnClickListener { tv_toDBC.text = toDBC(ed_string.text.toString()) }
        btn_toSBC.setOnClickListener { tv_toSBC.text = toSBC(ed_string.text.toString()) }
        btn_oneCn2ASCII.setOnClickListener { tv_oneCn2ASCII.text = oneCn2ASCII(ed_string.text.toString()).toString() }
        btn_oneCn2PY.setOnClickListener { tv_oneCn2PY.text = oneCn2PY(ed_string.text.toString()) }
        btn_getPYFirstLetter.setOnClickListener { tv_getPYFirstLetter.text = getPYFirstLetter(ed_string.text.toString()) }
        btn_getPYAllFirstLetter.setOnClickListener { tv_getPYAllFirstLetter.text = getPYAllFirstLetter(ed_string.text.toString()) }
        btn_cn2PY.setOnClickListener { tv_cn2PY.text = cn2PY(ed_string.text.toString()) }
    }

    override fun initData() {

    }

}